### PR TITLE
Update viewing of notes and appointment in patients

### LINKED ISSUE #92: Update UI to display appointment on patients
The appointment display will be similar to AB3's tag, with the following format:
<DESCRIPTION> Start: <START> End: <END>

### LINKED ISSUE #102: Update view command to display patient notes
To update the CommandResult column for view commands, and fields in Person card for *notes*

### PR DESCRIPTION
**Pull request by choonzies**
Closes #92 Closes #102 

This PR is currently a draft version and is not fully completed. It may be helpful for debugging purposes as the feature development proceed.

### CODE REVIEW

**Comment by choonzies** [comment]
**This PR is now mergable**. 🎉 

*Changes:*
1) Updated the labels for appointment fields
2) Removed diagnosis and medication fields in person card to reduce verbosity

*Future UI changes:*
1) The appointment labels design has room for improvement
2) The datetime string can be parsed to a more human readable string (in both commandresult box and personcard)

File: src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java [hunk 1]
```diff
@@ -32,8 +32,9 @@ public class AddAppointmentCommand extends Command {
             + "\n"
             + "Example: " + COMMAND_WORD + " 1 "
             + PREFIX_APPOINTMENT + " Surgery "
-            + PREFIX_START + " dd-MM-yyyy-HH-mm "
-            + PREFIX_END + " dd-MM-yyyy-HH-mm ";
+            + PREFIX_START + " 01-01-2001-01-00 "
+            + PREFIX_END + " 21-12-2024-23-59 "
+            + "\n Datetime is in the form of dd-MM-yyyy-HH-mm";
```
Comment (niclammm) [inline src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java:37]: Nice work to ensure better example is used

**Review by niclammm — COMMENTED** [review]
Overall great and straighforward code

File: src/main/java/seedu/address/logic/Messages.java [hunk 2]
```diff
@@ -45,7 +45,11 @@ public static String format(Person person) {
                 .append("\n Diagnosis: ")
                 .append(person.getDiagnosis())
                 .append("\n Medication: ")
-                .append(person.getMedication());
+                .append(person.getMedication())
+                .append("\n\n Notes: ")
+                .append(person.getNotes().toString().isEmpty() ? "-" : person.getNotes())
+                .append("\n\n Appointment: ")
+                .append(person.getAppointment() == null ? "-" : person.getAppointment().toString());
```
Comment (clarud) [inline src/main/java/seedu/address/logic/Messages.java:52]: Implementation looks good, maybe can extract "-" for all empty string in the future

File: src/main/java/seedu/address/model/appointment/Appointment.java [hunk 3]
```diff
@@ -13,9 +13,9 @@
  */
 public class Appointment {
 
-    public final String description;
-    public final LocalDateTime start;
-    public final LocalDateTime end;
+    private final String description;
+    private final LocalDateTime start;
+    private final LocalDateTime end;
```
Comment (clarud) [inline src/main/java/seedu/address/model/appointment/Appointment.java:18]: Good defensive programming

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 4]
```diff
@@ -52,9 +57,28 @@ public PersonCard(Person person, int displayedIndex) {
...
         ward.setText("Ward: " + person.getWard().value);
+        /*
         diagnosis.setText("Diagnosis: " + person.getDiagnosis().value);
         medication.setText("Medication: " + person.getMedication().value);
-        notes.setText(person.getNotes().value);
+        notes.setText("Notes: " + (person.getNotes().toString().isEmpty() ? "-" : person.getNotes().value));
+        */
+        // Law of Demeter is not maintained, however the purpose simply retrieves information through getters,
+        // and there may be zero risk of modifications.
+        if (person.getAppointment() != null) {
```
Comment (clarud) [inline src/main/java/seedu/address/ui/PersonCard.java:67]: Just a nit, could be extracted out into a method to improve reusability

**Review by clarud — COMMENTED** [review]
Generally LGTM

File: src/main/java/seedu/address/logic/Messages.java (hunk 2 above)
Comment (choonzies) [reply src/main/java/seedu/address/logic/Messages.java:52]: That's a great idea, we can look into that when we make diagnosis and medication non compulsory in `add command`, then it shows the "-" 😄

File: src/main/java/seedu/address/ui/PersonCard.java (hunk 4 above)
Comment (choonzies) [reply src/main/java/seedu/address/ui/PersonCard.java:82]: I agree! Have extracted it out accordingly to enhance SLAP 👍

**Pull request merged**